docs/notebooks/analysis/nullpoint.ipynb

```diff
@@ -0,0 +1,235 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "2474e8ad-eaa6-4054-a3d3-dc0d8f21e10b",
+   "metadata": {},
+   "source": [
+    "# Null Point Finder\n",
+    "\n",
+    "[Null Point Finder]: ../../api_static/plasmapy.analysis.nullpoint.rst\n",
+    "\n",
+    "[Null Point Finder] is a functionality that is designed to find and analyze 3D magnetic null point locations and structures using a trilinear method as described in [**Hynes:2007**](https://docs.plasmapy.org/en/latest/bibliography.html#id20)\n",
+    "In this notenook we will briefly cover how [Null Point Finder] utilizes the trilinear method in order to locate and classify the strucutres of magnetic null points. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "9b3731e3-1936-4b26-8a05-2130669fd074",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "\n",
+    "from pathlib import Path\n",
+    "\n",
+    "from plasmapy.analysis import nullpoint\n",
+    "\n",
+    "plt.rcParams[\"figure.figsize\"] = [10.5, 0.56 * 10.5]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "eec0e81c-bbf8-4e01-b121-114401996f7c",
+   "metadata": {},
+   "source": [
+    "## Contents:\n",
+    "\n",
+    "1. [How Null Point Finder works](#How-Null-Point-Finder-works)\n",
+    "    1. [Locating a null point](#Locating-a-null-point)\n",
+    "    1. [Classifying a null point](#Classifying-a-null-point)\n",
+    "1. [Running through examples](#Running_through-examples)\n",
+    "    1. [Uniform regular grid with a model function](#Uniform-regular-grid-with-a-model-function)\n",
+    "    1. [Arbitary Regular Grid](#Arbitary-regular-grid)"
```

Comment:
Thanks for adding in the table of contents!  One problem with the current way we do notebooks is that tables of contents are not automatically generated, so it ends up being an extra maintenance step to have to update it.  In any case, it still is worth the effort.